pkg/docker/config/config.go

```diff
@@ -55,7 +56,27 @@ var (
 
 // SetAuthentication stores the username and password in the auth.json file
 func SetAuthentication(sys *types.SystemContext, registry, username, password string) error {
+	credConfigs, err := sysregistriesv2.GetRegistryCredentialHelpers(sys, registry)
+	if err != nil {
+		return err
+	}
+	if len(credConfigs) > 0 {
+		if err := func(credConfigs []string, registry, username, password string) error {
+			for _, credConfig := range credConfigs {
+				if err = setAuthToCredHelper(credConfig, registry, username, password); err != nil {
```

Comment:
So this should (log? and) try the next helper if this one is not set up. Right now the second or any further helpers are never tried at all — and if any fail, the errors are silently ignored.

Reply:
Outstanding / depends on “helper installed but not set up” detection.